webextension/js/popup.js

```diff
@@ -129,13 +136,74 @@ browser.runtime.sendMessage({method: "queryIdentities"}).then(identities => {
   document.querySelector(".identities-list").appendChild(fragment);
 });
 
-document.querySelector("#edit-containers-link").addEventListener("click", () => {
-  browser.runtime.sendMessage({
-    method: "openTab",
-    url: "about:preferences#containers"
-  }).then(() => {
-    window.close();
+function showEditContainersPanel() {
+  browser.runtime.sendMessage({method: "queryIdentities"}).then(identities => {
+    let fragment = document.createDocumentFragment();
+
+    identities.forEach(identity => {
+      let tr = document.createElement("tr");
+      fragment.appendChild(tr);
+      tr.setAttribute("data-identity-cookie-store-id", identity.userContextId);
+      tr.classList.add("clickable");
+      tr.innerHTML = `
+        <td>
+          <div class="userContext-icon"
+            data-identity-icon="${identity.image}"
+            data-identity-color="${identity.color}">
+          </div>
+        </td>
+        <td>${identity.name}</td>
+        <td class="edit-container">
+          <img
+            title="Edit ${identity.name} container"
+            data-identity-cookie-store-id="${identity.userContextId}"
+            id="edit-${identity.userContextId}-container-icon"
+            src="/img/container-edit.svg"
+            class="icon edit-container-icon" />
+        </td>
+        <td class="remove-container" >
+          <img
+            title="Remove ${identity.name} container"
+            data-identity-cookie-store-id="${identity.userContextId}"
+            id="delete-${identity.userContextId}-container-icon"
+            class="icon delete-container-icon"
+            src="/img/container-delete.svg"
+          />
+        </td>`;
+
+      tr.addEventListener("click", e => {
+        if (e.target.matches(".edit-container-icon")) {
+          showPanel("#edit-container-panel");
+        } else if (e.target.matches(".delete-container-icon")) {
+          console.log(`clicked to delete ${identity.userContextId} container`);
+        }
+      });
+    });
+
+    document.querySelector("#edit-identities-list").innerHTML = "";
+    document.querySelector("#edit-identities-list").appendChild(fragment);
   });
+  showPanel("#edit-containers-panel");
+}
+
+document.querySelector("#add-container-link").addEventListener("click", () => {
+  showPanel("#edit-container-panel");
```

Comment:
The `#edit-container-panel` is for editing a **single** container. The `#edit-containers-panel` (i.e., plural) is for editing the *list* of containers.